core/src/main/java/org/infinispan/interceptors/impl/BaseStateTransferInterceptor.java

```diff
@@ -223,7 +225,7 @@ private Object handleReadCommandReturn(InvocationContext rCtx, VisitableCommand
       } else if (ce instanceof AllOwnersLostException) {
          if (trace)
             getLog().tracef("All owners for command %s have been lost.", cmd);
-         return null;
+         return rCommand.acceptVisitor(rCtx, LostDataVisitor.INSTANCE);
```

Comment:
Done, https://issues.jboss.org/browse/ISPN-7923

Don't know about the scattered cache, but for the regular cache, I think it might be simpler to have the distribution interceptor call into the `PartitionHandlingManager` and let it decide what to do when all the owners were lost (like `TxDistributionInterceptor` does for tx commands).